core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -15,7 +15,67 @@ import scala.concurrent.duration._
 
 /**
   * A stream producing output of type `O` and which may evaluate `F`
-  * effects. If `F` is [[Pure]], the stream evaluates no effects.
+  * effects.
+  * Or: a lazy sequence of computations that may evaluate `F` effects `F` and yield values of type `O`.
+  *
+  * Streams are purely functional: a `Stream[F, O]` is a description for a computation,
+  * much like `cats.effect.IO` or `scalaz.Task` are just descriptions of computations.
+  * The methods in the `Stream` class only calculate new descriptions from others.
+  *
+  * Streams are lazy, pull-based data sources. To '''evaluate''' a stream is to extract from it
+  * as many elements as a _consumer_ of the stream may need for the result.
+  *   by performing as many `F`-effectful computations as necessary to get those elements.
+  *   Thus, the effects in `F` are not evaluated unless and until they are required (pulled) by a stream consumer.
+  *
+  * - '''Laziness''': streams are lazily evaluated sources for effectful computations.
+  *   The operations on a Stream only evaluate a finite prefix,
+  *   a finite part of a whole stream, as needed to supply the requests of the stream _consumers_.
+  *   Thus, there can be infinite streams, or Streams based on
+  *   an unlimited source of information (e.g. a Signal that may be sampled with as frequently as desired).
+  *
+  * - '''Source''': a stream may simply be a fixed list of pre-defined F-effectful computations.
+  *   However, it can also be backed by an external data source (e.g. a network connection).
+  *   It can also be backed by a effectful computation that can be re-evaluated,
```

Comment:
"It can also be backed by" shows up twice here. Reword one.

Reply:
Done.

Reply:
Rewritten.